cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -184,6 +193,22 @@ public KubeTwillPreparer withStatefulRunnable(String runnableName,
     return this;
   }
 
+  @Override
+  public SecureTwillPreparer withIdentity(String runnableName, String identity) {
+    if (!twillSpec.getRunnables().containsKey(runnableName)) {
+      throw new IllegalArgumentException("Runnable " + runnableName + " not found");
+    }
+    // In Kubernetes, the identity represents the service account used to run the pod.
+    runnableServiceAccounts.put(runnableName, identity);
```

Comment:
Do we need identity per runnableName? Although existing code withStatefulRunnable does so, identity or service account is set at pod level instead of runnable level (which kind of maps to container). Having "runnableName" as an argument seems unnecessary and  could be misleading.

Reply:
The identity abstraction is defined independently of Kubernetes and should be applicable to all Twill runnables. I can imagine there would be a case in which multiple twill runnables will have different identities, even if Kubernetes itself does not support having multiple identities per runnable if we define a runnable as a container.

Reply:
This is k8s implementation of the interface and k8s doesn't support per-container-identity, it would be better to surface when caller attempts to set different identities. 

Alternatively,  we could  consider having two interfaces
withIdentity(defaultIdentity) => set default identity for all runnables in a twill application 
withIdnetity(runnable, identity) =>  allow per-runnable identity override. (unsupported in k8s impl)

Reply:
I agree that setting the values in a map might be a bit redundant, but I still feel that the interface should stay the same since other extension interfaces allow for setting specific runnable properties, and the interface should remain generic. I've gone ahead and removed the service account map.